[closes #5 closes #2] Fix tests + update docker image

JasoonS opened the pull request:
Pretend this is PR #3 - some weird stuff happened where updates to this branch weren't being detected after I made this repo public.

JasoonS committed:
Fix Formatting

JasoonS committed:
reset docker network

JasoonS committed:
try fix network

JasoonS committed:
[closes #2] Fix the tests - run in remote docker mode instead

JasoonS committed:
Dummy Test for circleci

JasoonS committed:
[#2] Fix and cleanup mistake changes

JasoonS committed:
[#2] Use published docker image as base container

JasoonS committed:
[#2] Use dockerhub images + add base image to repo

JasoonS committed:
[#2] Finish readme + cleanup

JasoonS committed:
Update start script to do correct checks

JasoonS committed:
Update readme with latest container/info + build container on test

JasoonS committed:
[experiment] see if circleci works better when docker continually prints something

JasoonS committed:
[experiment] try remove dependent IP from the ci-tests

JasoonS committed:
[#2] update errors with npm

JasoonS committed:
[closes #5] fix git submodule issues - (I hope circleci picks up on this)

coeniebeyers commented on .circleci/images/basic-docker-base/Dockerfile:
```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:17.10
```
Can we either use 16.04 or 18.04

coeniebeyers replied:
@JasoonS should be a quick change :)

JasoonS replied:
Yes, this code is a piece of history ;)

I think the idea from the retreat was to use the `admin-cli` (perhaps it needs a better name) base image as the base image for all our tests.

So I decided to delete this file now, the admin-cli image will be kept up to date instead, and has all the docker requirements.

Side note:

This will be extremely useful for our tests (the others much more than this one), and also debugging our circleci tests by using the `re-run with ssh` option in circleci. Then you can ssh into a machine with the cli tool already installed to help you work out what is wrong.

coeniebeyers replied:
Okay great!

JasoonS committed:
#5 - remove unused docker file - use the adhara-cli as base image

The admin-cli (adhara) took can be used for all of our tests which could be useful in the future as more features are added to admin-cli
